lib/internal/readline/interface.js

```diff
@@ -789,10 +809,41 @@ class Interface extends InterfaceConstructor {
 
   [kLine]() {
     const line = this[kAddHistory]();
+    this.undoStack = [];
+    this.redoStack = [];
     this.clearLine();
     this[kOnLine](line);
   }
 
+  [kPushToUndoStack](text, cursor) {
+    this.undoStack.push({ text, cursor });
+    if (this.undoStack.length > kMaxUndoRedoStackSize) {
+      this.undoStack.shift();
```

Comment:
```suggestion
      ArrayPrototypeShift(this.undoStack);
```

Reply:
Oops... I forgot to use the primordials again. Thanks!